packages/web-api-client/src/a11y-service-credential.ts

```diff
@@ -12,22 +13,33 @@ export class A11yServiceCredential {
         private readonly clientSecret: string,
         private readonly resourceId: string,
         authorityUrl: string,
+        private readonly logger: Logger,
         context?: AuthenticationContext,
+        private readonly maxTokenAttempts: number = 5,
+        private readonly sleepBetweenRetries: boolean = true,
     ) {
         // tslint:disable-next-line: no-any no-unsafe-any strict-boolean-expressions
         this.authContext = context || new (<any>AuthenticationContext)(authorityUrl, undefined, undefined, '');
     }
 
     public async getToken(): Promise<TokenResponse> {
-        return new Promise((resolve, reject) => {
-            this.authContext.acquireTokenWithClientCredentials(this.resourceId, this.clientId, this.clientSecret, (err, tokenResponse) => {
-                if (!isNullOrUndefined(err)) {
-                    reject(err);
-                } else {
-                    resolve(tokenResponse as TokenResponse);
+        await this.logger.setup();
+
+        let lastError: Error;
+        for (let i = 0; i < this.maxTokenAttempts; i += 1) {
```

Comment:
create a utility file, may be RetryHelper, that is generic & can be used in other places